hermes/src/network/p2p.rs

```diff
@@ -53,16 +53,41 @@ pub struct ObservationC {
     pub vaa_len: usize,
 }
 
-pub type Observation = Vec<u8>;
+/// A wrapper around a VAA observed from Wormhole.
+///
+/// This wrapper tracks a Span that allows tracking the VAA through the system. This Span is
+/// expected to be tied to the `proxy` Span and so logging will always be traced back to the
+/// associated `proxy` Span regardless of where in the system it is being used.
+#[derive(Clone, Debug)]
+pub struct Vaa {
```

Comment:
This is temporary, I think we probably want `struct Traceable<T> { span: Span, data: T }` to support the idea of traceable data more generically, this was just a quick way to test.